File: problems/p01/normalize_name.js
// [REQUIRED] 5–10 lines: how you understand the problem & approach,
// edge cases, and Big-O for time/memory.
function normalizeName(s) {
  const parts = s.trim().split(/\s+/).filter(Boolean);
  return parts.map(p => p.charAt(0).toUpperCase() + p.slice(1).toLowerCase()).join(" ");
}
module.exports = { normalizeName };

File: problems/p01/normalize_name.py
def normalize_name(s: str) -> str:
    """
    [REQUIRED] 5–10 lines: how you understand the problem & approach,
    edge cases, and Big-O for time/memory.
    """
    # 1) Split tokens (ignore empties caused by extra spaces)
    # 2) Title-case each token (first letter upper, rest lower)
    # 3) Join with a single space
    parts = [p for p in s.strip().split() if p]
    parts = [p[:1].upper() + p[1:].lower() for p in parts]
    return " ".join(parts)
